_rules/device-motion-disabled-c249d5.md

```diff
@@ -30,11 +30,11 @@ The rule applies to an [HTML document][] with an associated [Window object][] th
 
 ## Expectation 1
 
-For each registered [device orientation event][device orientation] or [device motion event][device motion] in the test target there exists a set of one or more [instruments][instrument] to prevent the results of the event.
+For each registered [device orientation event][device orientation] or [device motion event][device motion] in the test target there exists a set of one or more [instruments][instrument] to prevent any results of the event.
```

Comment:
Coming from a strong computability background, I'm always worried when we go into the effects of programs as essentially everything is undecidable, which make testing complicated… I'll try to flesh out concerns in a more concrete way…

* What if the instrument result in adding a 2h wait between the event trigger and the effect? This should fail the rule, but this is hardly testable. Maybe we need an arbitrary timeout on the test here also (as you did in another rule).

* What if (after activating the instrument), the effect of the event is to increase an internal counter which is never shown in any way to anybody? That would fail the arguably rule (depending what we call "result of the event") but not the SC.

* What if that counter can be seen but somewhere else? For example, that counter is stored on a server-side database and admins can look at it (but users can't). This is a bit more clearly a "result of the event", but still I'm not sure it fails the SC 

* Even more obvious, the event could both change a slider and update a counter, which is shown on a totally different page (e.g., profile page says "you've moved your phone X times"). And the instrument removes the update to slider on same page, but not the counter update. This wouldn't "prevent any result" but I'm not sure it fails the SC.

* From the previous points, if device motion update a counter that is only shown on another page, does this fail the SC? It might be very hard to test if we consider this as a result of the event (because it can't be tested in isolation). Maybe we need to restrict the rule to events that have an "effect" (whatever we mean by this) on the current page?

* Ultimately, even just `if (eventHasBeenDisabled) { return }` is some sort of "result of the event". Arguably, from a pure computability point of view, something is happening and we're consuming CPU time. Can be pushed to the absurd with `if (eventHasBeenDisabled) { doHugeComputation();  return }` which could grind the browser to almost a halt for 1 minute but ultimately have no other effect. Is stopping the browser a "result of the event"? This shows the ambiguity of these terms if the first case has no result but the second has (where is the limit?)


---
I think we need:
* a timeout on "nothing happens", otherwise this is undecidable.
* a more precise definition of the kind of things that can happen. This can totally start small as being too restrictive will mean that we sometimes pass the rule when the SC fails, which is OK.
I say something like "change in the DOM", or "change in the rendered pixels" is a good start for effect we want to monitor. It would mean we'll pass (or be inapplicable), say, a page where motion plays a sound, but that doesn't break the fail-on-fail relationship, so it only mean that the rule can be improved…

Reply:
@Jym77 I think the timeout is reasonable and I'll add it.

However, for the second issue I think you are missing one of the rule's assumptions that states: 
> The event listeners listening to device motion events trigger a functionality in the web page. If they do not trigger any such functionality failing this rule might not be a failure of the success criterion.

Do you agree that this addresses your concerns?

Reply:
That should be good, indeed.